src/blenderbim/blenderbim/bim/module/root/prop.py

```diff
@@ -88,6 +89,24 @@ def get_ifc_classes(self, context):
     return IfcClassData.data["ifc_classes"]
 
 
+def get_ifc_classes_suggestions():
+    if not IfcClassData.is_loaded:
+        IfcClassData.load()
+    suggestions = deepcopy(IfcClassData.data["ifc_classes_suggestions"])
+    file = IfcStore.get_file()
+    if file:
+        for ifc_class in get_ifc_classes(None, None):
+            ifc_class = ifc_class[0]
+            declaration = IfcStore.get_schema().declaration_by_name(ifc_class)
+            for attribute in declaration.attributes():
+                if attribute.name() == "PredefinedType":
+                    for e in attribute.type_of_attribute().declared_type().enumeration_items():
+                        if e in ("NOTDEFINED", "USERDEFINED",):
+                            continue
+                        suggestions[ifc_class].append(e)
```

Comment:
Can all this code be refactored into IfcClassData?

Reply:
Sure that makes sense. Thanks for the feedback ! :)